### DISCUSSION TITLE
Virtual field

### DISCUSSION THREAD

**Discussion by wowczarczyk** [react-hook-form/react-hook-form#2232] [Ideas] [answered]
Is there a way currently to mark a given field as virtual/transient? Let's say for instance "Accept terms" on a singup form. I would like this field to be part of the form (i.e. for validation) but I am not interested in sending this field to back end. I could obviously strip it off in the onSubmit method somewhere but maybe we could have an option of excluding it befor it is passed to `onSubmit`?

**Comment by bluebill1049** [comment, marked as answer]
Totally, https://react-hook-form.com/advanced-usage/#Workingwithvirtualizedlists you can register input at useEffect.

**Reply by Noitidart** [reply]
Should the useEffect run every render? or just on mount? so like:

no dep array:
```
useEffect(() => {
  register()
})
```

with mount dep array:
```
useEffect(() => {
  register()
}, [])
```

**Reply by bluebill1049** [reply]
```
useEffect(() => {
  register()
}, [register])
```

**Comment by bnheise** [comment]
I don't see how this answer answers the question. It certainly shows that we can register a component in use effect, but it doesn't provide any information about how to remove the field when we submit. In the example provided, the code sandbox example (https://codesandbox.io/s/react-hook-form-controlled-mixed-with-uncontrolled-forked-c323j) does indeed show registering, but when we click submit the registered field is still submitted. This is not the behavior that OP asked about.

**Comment by leapful** [comment]
@bnheise 

This discussion is really old and related to previous version of react-hook-form that might have different behavior than the latest version.

You can use `unregister` to perform exclude a field on submitting.

Working demo: https://codesandbox.io/s/react-hook-form-unregister-forked-pk39fj?file=/src/index.tsx

**Comment by bnheise** [comment]
Thanks for the tip!
